src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy

```diff
@@ -220,4 +221,107 @@ class SurrealPersistenceService implements PersistenceService {
         return result
     }
 
+    // get build count by specific metric ( ip, userName, targetImage)
+    @Override
+    LinkedHashMap<String, Long> getBuildsCountByMetric(Metric metric, MetricFilter filter){
+        def statement = "SELECT ${metric.buildLabel}, count() as total_count FROM wave_build "+
+                                "${getBuildMetricFilter(filter)} GROUP BY ${metric.buildLabel}  ORDER BY total_count DESC LIMIT $filter.limit"
+        final map = surrealDb.sqlAsMap(authorization, statement)
+        def results = map.get("result") as List<Map>
+        log.trace("Build count results by ${metric.buildLabel}: $results")
+        LinkedHashMap<String, Long> counts = new LinkedHashMap<>()
+        for(def result : results){
+            counts.put((result.get(metric.buildLabel)?:"unknown") as String, result.get("total_count") as Long)
```

Comment:
Let's replace `unknown` with `anonymous` and use a constant

Reply:
done